include/Dialect/Poly/IR/PolyPatterns.td

```diff
@@ -0,0 +1,33 @@
+#ifndef INCLUDE_DIALECT_POLY_IR_POLYPATTERNS_TD_
+#define INCLUDE_DIALECT_POLY_IR_POLYPATTERNS_TD_
+
+include "PolyOps.td"
+include "mlir/Dialect/Arith/IR/ArithOps.td"
+include "mlir/IR/OpBase.td"
+include "mlir/IR/PatternBase.td"
+
+// A hack to get the proper integer attribute type from the (possibly
+// non-power-of-two) ring coefficient modulus.
+// TODO: perhaps we can move this logic into the semantics of mul_scalar,
+// or else include it in the lowering.
```

Comment:
As in, mul_scalar verifies integer size is less than the coefficient modulus size, and then potentially also extui's it during lowering? Sounds fine with me!